services/payments/datastore.go

```diff
@@ -512,42 +356,103 @@ func (s *Service) AuthorizeTransaction(ctx context.Context, keyID string, transa
 	return nil
 }
 
-// GetTransactionFromDocID - get the transaction data from the document ID in qldb.
-func (s *Service) GetTransactionFromDocID(ctx context.Context, docID string) (*Transaction, error) {
-	transaction, err := s.datastore.Execute(context.Background(), func(txn qldbdriver.Transaction) (interface{}, error) {
-		resp := new(Transaction)
+// GetTransactionFromDocumentID - get the transaction data from the document ID in qldb.
+func (s *Service) GetTransactionFromDocumentID(
+	ctx context.Context,
+	documentID string,
+) (*AuthenticatedPaymentState, uuid.UUID, error) {
+	paymentStateInterface, err := s.datastore.Execute(
+		context.Background(),
+		func(txn qldbdriver.Transaction) (interface{}, error) {
+			resp := new(PaymentState)
+
+			result, err := txn.Execute(
+				"SELECT data.*, d_id FROM transactions BY d_id WHERE d_id = ?",
+				documentID,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get tx: %s due to: %w", documentID, err)
+			}
+			// Check if there are any results
+			if result.Next(txn) {
+				// unmarshal enriched version
+				err := ion.Unmarshal(result.GetCurrentData(), resp)
+				if err != nil {
+					return nil, fmt.Errorf("failed to unmarshal tx: %s due to: %w", documentID, err)
+				}
+			}
+
+			return resp, nil
+		},
+	)
+	if err != nil {
+		return nil, uuid.Nil, fmt.Errorf("failed to get transactions: %w", err)
+	}
+	paymentState := paymentStateInterface.(*PaymentState)
+	authenticatedState, err := s.PaymentStateToAuthenticatedPaymentState(ctx, *paymentState)
```

Comment:
we should pass the documentID to this function so it can verify it against any internal copies for consistency ( e.g. authorizations )